src/bytes_ser_de.rs

```diff
@@ -273,6 +275,223 @@ pub fn to_leb128_len(n: usize) -> usize {
     size
 }
 
+impl Serializable for bool {
+    type Error = CryptoCoreError;
+
+    fn length(&self) -> usize {
+        1
+    }
+
+    fn write(&self, ser: &mut Serializer) -> Result<usize, Self::Error> {
+        ser.write_leb128_u64(*self as u64)
+    }
+
+    fn read(de: &mut Deserializer) -> Result<Self, Self::Error> {
+        let b = de.read_leb128_u64()?;
+        match b {
+            0 => Ok(false),
+            1 => Ok(true),
+            _ => Err(CryptoCoreError::GenericDeserializationError(format!(
+                "not a valid boolean value serialization {b}"
+            ))),
+        }
+    }
+}
+
+impl Serializable for u64 {
+    type Error = CryptoCoreError;
+
+    fn length(&self) -> usize {
+        // Re-code `to_leb128_len` in order to avoid breaking change.
+        let mut n = *self >> 7;
+        let mut size = 1;
+        while n != 0 {
+            size += 1;
+            n >>= 7;
+        }
+        size
+    }
```

Comment:
very readable approach but how about the **math_approach** to optimise further ?

feel free to delete my comments

```suggestion
        // Re-code `to_leb128_len` in order to avoid breaking change.
        if *self == 0 {
            1
        } else {
            // 64 - self.leading_zeros() gives the number used bits
            // the +6 (basically... 7-1) is to ceil up the division
            // so what this does is simply replicate the previous while loop
            // but using arithmetic. The +6 accounts for the fact that when there is
            // a remainder, we need to add an extra byte.
            (64 - self.leading_zeros() as usize + 6) / 7
        }
```

Reply:
I didn't know about `div_ceil` , very good new approach !